src/sql/platform/accounts/common/accountActions.ts

```diff
@@ -50,7 +50,14 @@ export class AddAccountAction extends Action {
 		// Fire the event that we've started adding accounts
 		this._addAccountStartEmitter.fire();
 		try {
-			await this._accountManagementService.addAccount(this._providerId);
+			if (!this._providerId) {
+				this._providerId = await this._accountManagementService.promptProvider();
```

Comment:
Instead of setting this `providerId` - try using a local variable to fix the issue with provider Id being cached due to unresolved promise.